File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/abstractfactory/trasnport/Bicycle.java
package com.bigboxcode.designpattern.abstractfactory.trasnport;

public class Bicycle implements Transport {
    @Override
    public void start() {
        System.out.println("Bicycle Started");
    }

    @Override
    public void stop() {
        System.out.println("Bicycle Stopped");
    }

    @Override
    public void repair() {
        System.out.println("Bicycle Repair");
    }
}

File: design-patterns/ts/abstract-factory/transport/bicycle.ts
// bicycle.ts

import Transport from "./transport";

class Bicycle implements Transport {
    start(): void {
        console.log("Bicycle Started");
    }
    stop(): void {
        console.log("Bicycle Stopped");
    }
    repair(): void {
        console.log("Bicycle Repair");
    }
}


export default Bicycle;